### PR TITLE
feat: set fsspec as default source

### PR DESCRIPTION
**Pull request by lobis**
- Removes support for the `*_handler` arguments (`s3_handler`, `file_handler`, etc.) which were already on a deprecation notice.

- Sets the fsspec source as default for all cases except:
    - file-like objects 
    - s3 due to https://github.com/scikit-hep/uproot5/pull/1012 (if set as default pytest will fail)

### CODE REVIEW

File: tests/test_0017_multi_basket_multi_branch_fetch.py [hunk 1]
```diff
@@ -310,6 +310,7 @@ def test_cache():
         skhep_testdata.data_path("uproot-sample-6.20.04-uncompressed.root"),
         object_cache=100,
         array_cache="100 MB",
+        handler=uproot.source.file.MemmapSource,
```
Comment (lobis) [inline tests/test_0017_multi_basket_multi_branch_fetch.py:313]: Not so sure I am not breaking anything here, should the `fsspec` source behave exactly the same as the previous default source?

File: tests/test_0173_empty_and_multiprocessing_bugs.py [hunk 2]
```diff
@@ -17,8 +16,8 @@ def test_empty():
         assert t["z"].array(library="np").tolist() == []
 
 
-def readone(filename):
-    with uproot.open(filename) as f:
+def read_one(filename):
+    with uproot.open(filename, handler=uproot.source.file.MemmapSource) as f:
```
Comment (lobis) [inline tests/test_0173_empty_and_multiprocessing_bugs.py:20]: https://github.com/scikit-hep/uproot5/issues/173

This won't work with fsspec at the moment (meaning it won't work without manually specifing a different source), I'm not sure what the consecuences might be

File: tests/test_0017_multi_basket_multi_branch_fetch.py (hunk 1 above)
Comment (jpivarski) [reply tests/test_0017_multi_basket_multi_branch_fetch.py:313]: If the default file-reader becomes non-memory mapping, that's okay. `MemmapSource` is still opt-in, and memory-mapping wasn't as important of a performance feature as it was originally thought to be. (The data have to be decompressed, byte-swapped, and concatenated, so the data in the file can't be used directly, anyway.)

File: tests/test_0173_empty_and_multiprocessing_bugs.py (hunk 2 above)
Comment (jpivarski) [reply tests/test_0173_empty_and_multiprocessing_bugs.py:20]: From the looks of it, this test was going after a bug that's particular to memory-mapping. Explicitly setting the handler to recreate that situation is fine.

**Review by jpivarski — APPROVED** [review]
Good—this is going into `main-fsspec`, to be part of the 5.2.0rc1 pre-release.

It's ready to merge into that (non-default) branch.

**Review by nsmith- — COMMENTED** [review]
I suppose we should ask @eduardo-rodrigues based on his input to #692 
For me it seems to deserve a minor version bump, otherwise OK.

**Comment by nsmith-** [comment]
Well, I guess this is into `main-fsspec`, so what is the path to merge `main-fsspec` into `main`?

**Comment by lobis** [comment]
> Well, I guess this is into `main-fsspec`, so what is the path to merge `main-fsspec` into `main`?

- https://github.com/scikit-hep/uproot5/pull/1022

It'll be merged when 5.2.0 is released (decemeber?) but in the meantime we will publish rc from this branch

**Comment by eduardo-rodrigues** [comment]
> I suppose we should ask @eduardo-rodrigues based on his input to #692 For me it seems to deserve a minor version bump, otherwise OK.

Hello. Sorry to come late to the party. Sounds good, @nsmith- what you propose anyway.

**Pull request merged**